be/src/storage/vectorized/column_predicate_cmp.cpp

```diff
@@ -0,0 +1,789 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include <cstdint>
+#include <functional>
+#include <utility>
+#include <vector>
+
+#include "column/chunk.h"
+#include "column/column.h" // Column
+#include "column/datum.h"
+#include "common/object_pool.h"
+#include "storage/olap_common.h" // ColumnId
+#include "storage/rowset/bitmap_index_reader.h"
+#include "storage/rowset/bloom_filter.h"
+#include "storage/types.h"
+#include "storage/vectorized/column_predicate.h"
+#include "storage/vectorized/range.h"
+#include "storage/vectorized/zone_map_detail.h"
+#include "util/string_parser.hpp"
+
+namespace starrocks {
+class BloomFilter;
+class Slice;
+class ObjectPool;
+class ExprContext;
+class RuntimeState;
+class SlotDescriptor;
+class BitmapIndexIterator;
+class BloomFilter;
+} // namespace starrocks
+
+namespace starrocks::vectorized {
+
+template <FieldType field_type>
+using GeEval = std::greater_equal<typename CppTypeTraits<field_type>::CppType>;
+
+template <FieldType field_type>
+using GtEval = std::greater<typename CppTypeTraits<field_type>::CppType>;
+
+template <FieldType field_type>
+using LeEval = std::less_equal<typename CppTypeTraits<field_type>::CppType>;
+
+template <FieldType field_type>
+using LtEval = std::less<typename CppTypeTraits<field_type>::CppType>;
+
+template <FieldType field_type>
+using EqEval = std::equal_to<typename CppTypeTraits<field_type>::CppType>;
+
+template <FieldType field_type>
+using NeEval = std::not_equal_to<typename CppTypeTraits<field_type>::CppType>;
+
+template <FieldType field_type, template <FieldType> typename Predicate, typename NewColumnPredicateFunc>
```

Comment:
ColumnPredicateFunc may be a better name.

Reply:
done.